sdk/spring/spring-reference.yml

```diff
@@ -1,428 +1,349 @@
-spring:
-  - name: Azure Spring Boot Bom
-    content:
-      - name: Azure Spring Boot Bom
-        id: spring-boot-bom
-        description: BOM for Microsoft Azure Spring Boot support.
-        artifacts:
-          - artifactId: azure-spring-boot-bom
-            groupId: com.azure.spring
-            versionGA: 3.2.0
-            versionPreview: 3.0.0-beta.1
-            bom:
+- name: Azure Spring Boot Bom
+  content:
+    - name: Azure Spring Boot Bom
+      id: spring-boot-bom
+      description: BOM for Microsoft Azure Spring Boot support.
+      artifacts:
+        - artifactId: azure-spring-boot-bom
+          groupId: com.azure.spring
+          versionGA: 3.2.0
+          versionPreview: 3.0.0-beta.1
+          type: spring
+          links:
+            repopath: https://search.maven.org/artifact/com.azure.spring/azure-spring-boot-bom/3.2.0/jar/
+            github: https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/boms/azure-spring-boot-bom
+          springProperties:
             compatibilityRange: "[2.2.11.RELEASE,2.4.0-M1)"
             mappings:
               - compatibilityRange: "[2.0.0.RELEASE,2.3.0.M1)"
                 groupId: com.microsoft.azure
                 artifactId: azure-spring-boot-bom
-            scope:
-            alias:
-            facets:
-            starter: false
-            links:
-              - rel: repopath
-                href: https://search.maven.org/artifact/com.azure.spring/azure-spring-boot-bom/3.2.0/jar/
-              - rel: msdocs
-                href:
-              - rel: ghdocs
-                href:
-  - name: Active Directory
-    content:
-      - name: Active Directory
-        id: active-directory
-        description: Azure Active Directory (Azure AD) is Microsoft’s cloud-based identity and access management service, which helps your employees sign in and access resources.
-        artifacts:
-          - artifactId: azure-spring-boot-starter-active-directory
-            groupId: com.azure.spring
-            versionGA: 3.2.0
-            versionPreview:
+                version: 2.2.4
+- name: Active Directory
+  content:
+    - name: Active Directory
+      id: active-directory
+      description: Azure Active Directory (Azure AD) is Microsoft’s cloud-based identity and access management service, which helps your employees sign in and access resources.
+      msdocs: https://docs.microsoft.com/azure/active-directory/fundamentals/
+      artifacts:
+        - artifactId: azure-spring-boot-starter-active-directory
+          groupId: com.azure.spring
+          versionGA: 3.2.0
+          type: starter
```

Comment:
how about making `starter` a subtype of `spring` and distinguish `starter` from `vanilla spring` by a flag in `springProperties`:

```
type: spring
...
springProperties:
    starter: true
```